core/camera/src/main/java/com/google/jetpackcamera/core/camera/CameraXCameraSystem.kt

```diff
@@ -500,37 +553,360 @@ constructor(
             }
     }
 
-    private fun resolveStabilizationMode(
+    /**
+     * Updates the [CameraSystemConstraints] based on feature group compatibility.
+     *
+     * This function checks various combinations of settings (dynamic range, frame rate,
+     * stabilization, etc.) against the device's capabilities using the CameraX feature groups API.
+     * It filters out unsupported options from the system constraints, ensuring that the UI only
+     * presents valid combinations to the user.
+     *
+     * This update happens asynchronously after the initial camera session is started to avoid
+     * blocking the UI thread.
+     */
+    private suspend fun updateSystemConstraintsByFeatureGroups() {
```

Comment:
Is there any way we can move this to a seaparate file. CameraXCameraSystem is in dire need of refactoring, but I'd prefer not to add so many lines of code if possible.

Eventually I would like to move all of the constraints logic out, which may even help with consistency in the Settings screen.